docs/decisions/0004-axios-caching-implementation.rst

```diff
@@ -0,0 +1,87 @@
+In order to keep the interface more simple the cached clients don’t have their own getter functions exposing this detail, in order to use the cached http client micro frontends pass in a param for using the cache with the current getter functions.
+
+In the case of the cached clients failing to get created the uncached clients will be returned and used instead.
+
+The async hydrateAuthenticatedUser function will continue only ever using the uncached authenticated HTTP client so there won't be any concerns about the cache changing expected behavior of authentication.
+
+Only the AxiosJwtAuthService currently has cached http clients.
+
+Decisions
+---------
+
+By using the existing axios-cache-adapter we were able to avoid needing to implement our own caching strategy and forgo having micro frontends need to implement their own front end caching strategies.
+
+The default max-age for cached data was decided to be 5 minutes for now. The decision was based on it being a decent amount of the storage of cached data even for authentication specific requests. The assumption is made that sers won’t be frequently switching to different accounts on their browser frequently enough to have a wide impact and if they do then they will only see cached data for a maximum of 5 minutes. Micro frontends are also able to override the cache time to suit their own needs.
+The default max-age can also be changed at a later time if the current default of 5 minutes is deemed to be inadequate.
+
+Tests written to verify the cache adapter integrates well with the current interceptors make real external API calls to https://httpbin.org/ as per the tests in the axios-cache-adapter source code.
+It was found to be impossible to mock out the exact spot the cache client makes an outside request without also mocking over the cache clients caching functionality. This wasn’t the preferred approach to testing because it makes external network requests but since it is the approach that was used in the axios-cache-adapter tests it was decided to be fine.
+
+Alternatives
+------------
+
+We discussed the pros and cons of having a front-end platform cache implementation for micro front ends to use versus micro front ends implementing their own front end caching strategies utilizing Local/Session storage, ETags, and endpoints utilizing Cache-Control headers.
+
+Adoption
+--------
+
+As of this writing, only a few HTTP requests within frontend-app-learner-portal-enterprise have adopted using the cached clients. Usage of the cache clients is on an opt-in basis where micro frontends determine whether or not using the front end cache clients suits their needs.
+
+
+Consequences
+------------
+
+Axios is version locked to version 0.18.1 so per request overrides work until this issue gets resolved: https://github.com/RasCarlito/axios-cache-adapter/issues/99.
```

Comment:
Does this mean the per-request overrides (like `{ useCache: true }`) won't work on later versions?

Reply:
That's the service function for getting the httpClient. The per-request overrides are the GET, POST, ETC (like `client.get('https://google.com', {cache: {maxAge: 1000}})` requests made using the client instance. Overrides like deciding to use a different storage or the maxAge of the cache data. Here's all of the options: https://www.npmjs.com/package/axios-cache-adapter#options

I think the issue is fixed or resolved in Axios 0.20 so I think at worst we might just need to skip versions between 0.18.1 and 0.20

Reply:
👍 We should probably make a note about that in here or in `package.json`.